packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -77,8 +77,8 @@ export interface RenderAPI {
     ): VNode;
     i(items: any[], factory: () => VNode | VNode): VNodes;
     f(items: any[]): any[];
-    t(text: string): VText;
-    d(value: any): VNode | null;
+    t(text: string, data: ElementCompilerData): VText;
```

Comment:
Should this be called NodeCompilerData? It would be non-intuitive to generate ElementCompilerData for a text node. 
Although, ElementCompilerData and its parent VNodeData seems bloated for a text node.
```suggestion
    t(text: string, data: NodeCompilerData): VText;
```

Reply:
in theory it could be just "key", i made it an object so it's compatible with the existing API (and if extended in the future the api does not need to change).

What about:
```js
export interface CompilerData {
    key: Key;
}

export interface ElementCompilerData extends CompilerData, VNodeData {}
```

and use `CompilerData`?